accounting/common/logging.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2017-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+Logging for the accounting service.
+"""
+
+from logging.handlers import RotatingFileHandler
+import gzip
+import os
+
+
+class GzipRotatingFileHandler(RotatingFileHandler):
+    """
+    Uses gzip to compress the rotated log files.
+    https://docs.python.org/3/howto/logging-cookbook.html#using-a-rotator-and-namer-to-customize-log-rotation-processing
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(GzipRotatingFileHandler, self).__init__(*args, **kwargs)
+        self.namer = self._namer
+        self.rotator = self._rotator
+
+    @staticmethod
+    def _namer(name):
+        """Append a .gz suffix to the rotated files."""
+        return name + '.gz'
+
+    @staticmethod
+    def _rotator(source, dest):
+        """Write compressed output when rotating files."""
+        with open(source, "rb") as sf:
+            with gzip.open(dest, "wb", compresslevel=9) as df:
```

Comment:
This is copy and paste from https://github.com/open-craft/opencraft/blob/master/opencraft/logging.py. In order to ease maintenance between the two, I'd prefer to not touch this file. It might be a better long-term approach to move something like this out to a new, small library, uploaded to pip, but not sure.

Reply:
But we can certainly change it here *and* there if needed, however I didn't even read this code since I just knew what it did so I didn't bother fixing its readability.

Reply:
@UmanShahzad Ah, OK -- I wasn't aware that we had the same piece of code in OCIM. Definitely fine to keep this as is, then :)

Reply:
@UmanShahzad Copy-pasting code is a pretty bright red flag - I don't think we should do this. Could you move that code to a repository containing shared utils code like this, which you then add to the requirements on both sides?

Reply:
@antoviaque Yeah I think it'd be good to have a common repo for these kinds of things across our application repositories. That's going to be quite some work though so I'll make a ticket. :+1:

Reply:
Created https://tasks.opencraft.com/browse/OC-3703.